### ISSUE TITLE
feature req: disable registration for agents also, registration only by administrators

### ISSUE THREAD

**Issue by mckaygerhard** [osTicket/osTicket-1.8#1665]
there 3 types of user access, disabled (all guest permited), public (any email, even fakes) and by agents (show as private)

a more "intranet like" access will be usefully for agents that are only for take phone calls, i mean a level when the agent cant registrer an user, and only admin users can do..

example:
Registration Method:	  admins only (agents cannot create users)

of course, this level assume that users must input a valid email for view their tickets.

if this feature request cannot be done, there some tips for make this for me?

**Comment by greezybacon** [comment]
Sounds like a permission for the new "roles" feature in v1.9.6. @protich?

**Comment by protich** [comment]
:+ it's planned for v1.9.6

**Comment by mckaygerhard** [comment]
great! weantime i remove temporally the button in user search for agents

**Comment by mckaygerhard** [comment]
hello @protich and @greezybacon ; any progress on that? what brand are using for that feature? can i made tests and to see code review?

**Comment by greezybacon** [comment]
@mckaygerhard the new roles feature is in the `develop-next` branch if you wanna check it out

**Comment by mckaygerhard** [comment]
there's instrucctions details how to upgrade manually (not automatically)?

**Comment by greezybacon** [comment]
@mckaygerhard it's not rocket surgery. Copy your database, unpack `develop-next` to another location. Copy your `include/ost-config.php` file into the new location. Edit `include/ost-config.php` to use the copied database. Visit the site and proceed with the data migration.

**Comment by mckaygerhard** [comment]
jeje u right, but the deploy procedure its tedius.. ok gimme 3 days from now and i'll report feedback ok? i must setup with that i was thinking for usage purpose..

**Comment by mckaygerhard** [comment]
i have some days testing the brand, yesterday i got that error:

```
Fatal error: Can't inherit abstract function AuthenticatedUser::getId() (previously declared abstract in BaseAuthenticatedUser) in /opt/simpleticket/simpleticket-1.9.6.7/include/class.auth.php on line 26
```

i reaintaled and the backported the database dumb... 

a hint: the update process does not work, i installed from scrat and works well .. not tested deeple but from now i'll backport to one of my production systems

**Comment by mckaygerhard** [comment]
i toggle today the current lasted from developmen-next and got this at new install:

```
Fatal error: Call to a member function getPrimaryLanguage() on a non-object in /opt/simpleticket/simpleticket-1.9.6.7/include/class.i18n.php on line 360
```

i delete the older tarring and i cannot install now! i cant remember from what commit i download the previously tar file

**Comment by ntozier** [comment]
This issue has veered far off topic.  Please open another issue report for your second issue.

**Comment by mckaygerhard** [comment]
ok, but i think its not property and issue, due its over a dvelop brand and not the official! ?  i must report anyway? its a problem over a non-versioned or valid release! either the normal brand!

**Comment by ntozier** [comment]
This issue report thread is for a feature request (which you know since it's yours). Your last several posts appear to be about your issues upgrading/installing, which has nothing to do with the feature request.

**Comment by mckaygerhard** [comment]
ah but if i report the error, will be not agains a stable release! i must report it and dont care if its not agains the master brand?

**Comment by ntozier** [comment]
I have no idea what you are trying to say.  master brand?  develop-next is develop next.  Its the place where all new code goes prior to being introduced into core.

**Comment by mckaygerhard** [comment]
ok, so in conclusion i cannot test the feature!

**Comment by ntozier** [comment]
Sure you can.  Download the develop-next branch and perform a new installation.

**Comment by mckaygerhard** [comment]
8 minutes ago, i clone the git repo, and changed the environment to the recomended: apache2, php 5.3.28, mysql 5.3.11 and again same error.. i goin to the error source and the file looks good..  the machine are a DELL T110 i3 4 cores 4G RAM dedicated (a machine used only for that purpose)

**Comment by mckaygerhard** [comment]
no luck, in code i noted that the cfg object are empty, so the come i suggested must be:

```
return $cfg ? $cfg->getPrimaryLanguage() : 'en_US';
```

in the line 360 of file class.i18n.php solves the problem, due the configuration object its not loaded yet  due its not installed! uff

**Comment by mckaygerhard** [comment]
now this error i must paste here, its related to this feature:

```
Fatal error: Can't inherit abstract function AuthenticatedUser::getId() (previously declared abstract in BaseAuthenticatedUser) in /var/www/osticket/include/class.auth.php on line 26
```

installation canot continue.. upgrade can be performed (or i confused the tarballs) but now not!

**Comment by mckaygerhard** [comment]
som more errors, as can see in issue #1984 seems that languaje support changed a lot and its very unstable..  

so cannot test this feeature until the cahnges that i comment are fixeds (i provide in the comments how to fix also ok)

**Comment by mckaygerhard** [comment]
hi? what the progress of that feature?

**Comment by ntozier** [comment]
![capture](https://cloud.githubusercontent.com/assets/1315768/7968228/5eeb41ea-09fd-11e5-9322-74646dd366f2.PNG)
They told you it was in develop-next branch on Mar 16th.
I'd say that it was done at that time.

**Comment by mckaygerhard** [comment]
but lasted revisions dont install there too many issues to test the developer-next.. i report some solutions and workarounds but i not get any noise or response in the issues

**Comment by harrisyn** [comment]
I do think he deserves a straight answer. Yes/no or soon. We have had at least 3 releases since he first raised it.

**Comment by mckaygerhard** [comment]
i revised the isue about stabilize the 1.10 code (based on develop-netx) and i not see directly this feature.. only references to the new role feature but into that feature there's no agents administration by admin roles.. only users administration

also.. from that comment "if u wanna check out" every time i try to clone and test, the brand has some problems in the code changes.. mostly with the i18n related code..

i made some reports and code solutions but i cannot see any response...or any commit related.

**Comment by mckaygerhard** [comment]
ping? devel brand still bronken in i18n related code.. unninstallable for now

**Comment by greezybacon** [comment]
@mckaygerhard I think you're mistaken. It installs just fine

**Comment by mckaygerhard** [comment]
so then, retest again, i'll post feedback when back

**Comment by mckaygerhard** [comment]
closed, profile are present in 1.10 series!
